request.js

```diff
@@ -74,14 +74,14 @@
             request.onload = function(){
                 if((request.statusText === 'OK' && request.status === 200) || typeof request.statusText === 'undefined'){
                     methods.success.apply(methods, utils.parse(request));
-                    methods.always.apply();    
+                    methods.always.apply();
                 } else {
                     methods.error.apply(methods, utils.parse(request));
                     methods.always.apply();
                 }
             };
-            request.onerror = function(){
-                methods.error.apply(methods, utils.parse(request));
+            request.onerror = function(e){
+                methods.error.apply(methods, e);
```

Comment:
My point is that `request` would not have much information at that point because of the failure, you have all you must to know in here:

```js
.get(url).error(function(e){
    OlapicSDK.debug.addError(e.error);
}
```